### DISCUSSION TITLE
Does Perseus support custom template frameworks

### DISCUSSION THREAD

**Discussion by XAMPPRocky** [framesurge/perseus#275] [Q&A]
Hello 👋 as part of project I'm currently considering adding a custom backend framework, however I'm mostly interested in controlling the presentation and data sources of the templates, beyond that it would have a lot of overlap with a project like Perseus. So I'm wondering if there is a way to have it so that instead of reimplementing the wheel, I could mostly control the parts I care about, and translate them in a way that would make it usable by Perseus.

My question is, if I use Perseus as a library, can I generate Perseus apps, templates, and capsules entirely dynamically at runtime?

**Comment by arctic-hen7** [comment]
This is a feature that I am working on in a generic manner for v0.5 actually! In the meantime, it should certainly be *possible* with v0.4, especially with Sycamore's builder syntax or with direct HTML inclusions. It'll be a bit tricky for now, and the next version of Perseus will support this kind of scripting as a first-class feature, but all the best if you go ahead, and I'd love to see how it goes!
